perun-core/src/main/java/cz/metacentrum/perun/core/implApi/AttributesManagerImplApi.java

```diff
@@ -213,6 +213,18 @@
 	List<Attribute> getAttributes(PerunSession sess, Member member, Resource resource, List<String> attrNames) throws InternalErrorException;
 
 	/**
+	 * Get all attributes (empty and virtual too) associated with the member on the resource which have name in list attrNames.
```

Comment:
member on resource -> group on resource
also I believe empty list means "all non-empty attributes" which is not mentioned in javadoc.